File: src/pixelpast/api/schemas/social_graph.py
"""Schemas for social-graph projection payloads."""

from __future__ import annotations

from pydantic import BaseModel


class SocialGraphPersonGroup(BaseModel):
    """Matching person-group metadata exposed for one social-graph node."""

    id: int
    name: str
    color_index: int | None = None


class SocialGraphPerson(BaseModel):
    """Person node exposed by the social-graph API contract."""

    id: int
    name: str
    occurrence_count: int
    matching_groups: list[SocialGraphPersonGroup]


class SocialGraphLink(BaseModel):
    """Weighted unordered person-pair edge in the social graph."""

    person_ids: list[int]
    weight: int
    affinity: float


class SocialGraphResponse(BaseModel):
    """Canonical social-graph projection response."""

    persons: list[SocialGraphPerson]
    links: list[SocialGraphLink]

File: ui/src/projections/socialGraph.ts
export type SocialGraphPersonGroupProjection = {
  id: string;
  name: string;
  colorIndex: number | null;
};

export type SocialGraphPersonProjection = {
  id: string;
  name: string;
  occurrenceCount: number;
  matchingGroups: SocialGraphPersonGroupProjection[];
};

export type SocialGraphLinkProjection = {
  personIds: [string, string];
  weight: number;
  affinity: number;
};

export type SocialGraphProjection = {
  persons: SocialGraphPersonProjection[];
  links: SocialGraphLinkProjection[];
};
